book/internal-services.md

```diff
@@ -0,0 +1,83 @@
+| PipelineRun                 | Once a InternalRequest has been reconciled and the Pipeline to run has been discovered and is ready for execution | To perform the steps in the Internal Pipeline  |
+
+### READ
+
+| Custom Resources     | When?                                                       | Why?                                                                   |
+|----------------------|-------------------------------------------------------------|------------------------------------------------------------------------|
+| Config               | During controller startup and during each reconcile attempt | To obtain configuration information                                    |
+
+### UPDATE
+
+| Custom Resource  | When?                                                              | Why?                                                                      |
+|------------------|--------------------------------------------------------------------|---------------------------------------------------------------------------|
+| InternalRequest  | During the lifecycle of an attempt to complete an InternalRequest  | To provide status for the execution of the Pipeline to the remote cluster |
+
+### WATCH
+
+| Custom Resource            | When?                                                      | Why?                                                                               |
+|----------------------------|------------------------------------------------------------|------------------------------------------------------------------------------------|
+| InternalRequest            | Always                                                     | To provide an API to process an internal request                                   |
+| PipelineRun                | Once the PipelineRun is created                            | To relay the Internal PipelineRun status to the remote InternalRequest for viewing |
+
+## Security Risk Mitigations
+
+Enabling remote clusters the ability to run internal jobs carrying certain security risks. The following list describes measures in places to mitigate those risks.
+
+* The creation of an `InternalRequest` custom resource requires permission on the Remote Cluster.
+* The Internal Services controller instance is configured to watch a specific cluster.
+  * This cluster is provided to the controller as an argument to a secret that was added by the admin team.
+  * The secret contains a KUBECONFIG file.
+* Only pipelines that are defined within the controller's namespace can be executed on the Internal, Private cluster.
+  * The deployment of these pipelines is controlled by App Interface and is orchestrated via GitHub pull requests.
+  * In addition, pipelines deployed to production are pinned to Git commit SHAs described in App Interface.
```

Comment:
These are important for "our" use of the controller but they feel like implementation details that are more relevant for a threat model than a general overview. 

I feel like a more appropriate further refinement of this might be along the lines of tightly limiting the actions of the pipeline to reduce any potential harm due to arbitrary code execution.

Reply:
saved those bullets for the Threat Model.